client/blocks/reader-share/index.jsx

```diff
@@ -70,73 +84,75 @@ function buildQuerystringForPost( post ) {
 	return qs.stringify( args );
 }
 
-const ReaderShare = React.createClass( {
+function canShareToWordPress() {
+	return !! user.get().primarySiteSlug;
+}
+
+class ReaderShare extends React.Component {
 
-	propTypes: {
+	static propTypes = {
 		iconSize: React.PropTypes.number
-	},
+	}
 
-	getInitialState() {
-		return { showingMenu: false };
-	},
+	static defaultProps = {
+		position: 'bottom',
+		tagName: 'li',
+		iconSize: 24
+	}
 
-	getDefaultProps() {
-		return {
-			position: 'top',
-			tagName: 'li',
-			iconSize: 24
-		};
-	},
+	state = {
+		showingMenu: false,
+	};
 
 	componentWillUnmount() {
-		if ( this._closeHandle ) {
-			clearTimeout( this._closeHandle );
-			this._closeHandle = null;
+		if ( this.closeHandle ) {
+			clearTimeout( this.closeHandle );
+			this.closeHandle = null;
 		}
-	},
+	}
 
-	_deferMenuChange( showing ) {
-		if ( this._closeHandle ) {
-			clearTimeout( this._closeHandle );
+	deferMenuChange = ( showing ) => {
+		if ( this.closeHandle ) {
+			clearTimeout( this.closeHandle );
 		}
 
-		this._closeHandle = defer( () => {
-			this._closeHandle = null;
+		this.closeHandle = defer( () => {
+			this.closeHandle = null;
 			this.setState( { showingMenu: showing } );
 		} );
-	},
+	}
 
-	toggle( event ) {
+	toggle = ( event ) => {
 		event.preventDefault();
 		if ( ! this.state.showingMenu ) {
-			const target = !! sitesList.getPrimary() ? 'wordpress' : 'external';
+			const target = canShareToWordPress() ? 'wordpress' : 'external';
 			stats.recordAction( 'open_share' );
 			stats.recordGaEvent( 'Opened Share to ' + target );
 			stats.recordTrack( 'calypso_reader_share_opened', {
 				target
 			} );
 		}
-		this._deferMenuChange( ! this.state.showingMenu );
-	},
+		this.deferMenuChange( ! this.state.showingMenu );
+	}
 
-	closeMenu() {
+	closeMenu = () => {
 		// have to defer this to let the mouseup / click escape.
 		// If we don't defer and remove the DOM node on this turn of the event loop,
 		// Chrome (at least) will not fire the click
 		if ( this.isMounted() ) {
```

Comment:
using `isMounted` is deprecated and produces a React warning in the console.

<img width="1432" alt="screen shot 2017-04-10 at 10 56 15 am" src="https://cloud.githubusercontent.com/assets/4656974/24867879/8a29a370-1ddc-11e7-8d87-43ebf68816d2.png">